sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2584,6 +2586,42 @@ gather_impl(AccessorTy acc, simd<uint32_t, N> offsets, uint32_t glob_offset,
   }
 }
 
+template <typename T, int N, int VS, cache_hint L1H, cache_hint L2H,
+          lsc_data_size DS, typename OffsetT, typename AccessorT>
+__ESIMD_API std::enable_if_t<
+    is_device_accessor_with_v<AccessorT, accessor_mode_cap::can_read>,
+    simd<T, N>>
+gather_impl(AccessorT acc, simd<OffsetT, N / VS> byte_offsets,
+            simd_mask<N / VS> pred, simd<T, N> pass_thru) {
+  static_assert(N / VS >= 1 && N % VS == 0, "N must be divisible by VS");
+  static_assert(std::is_integral_v<OffsetT>,
+                "Gather must have integral byte_offset type");
+  static_assert(sizeof(OffsetT) <= 4,
+                "Implicit truncation of 64-bit byte_offset to 32-bit is "
+                "disabled. Use -fsycl-esimd-force-stateless-mem or explicitly "
+                "convert offsets to 32-bit vector");
+  static_assert(VS == 1 || sizeof(T) >= 4,
```

Comment:
In general Yes, but I think it is not required for this PR. If start assuming sizeof(T) may be greater than 8, then we'll need lots and lots of places to be fixed, not only this one.
For now, we don't have any type `T` that would be usable in `simd<T, N>` class, it will fail on attempt to create `simd` object.

Reply:
If the simd class will fail then we can drop this